modules/eks/datadog-agent/main.tf

```diff
@@ -40,21 +33,59 @@ locals {
       ]
     })
   }
-  set_datadog_cluster_checks = [for cluster_check_key, cluster_check_value in local.datadog_cluster_checks :
-    {
+  set_datadog_cluster_checks = [
+    for cluster_check_key, cluster_check_value in local.datadog_cluster_checks : {
       # Since we are using json pathing to set deep yaml values, and the key we want to set is `something.yaml`
       # we need to escape the key of the cluster check.
       name  = format("clusterAgent.confd.%s", replace(cluster_check_key, ".", "\\."))
       type  = "auto"
       value = yamlencode(cluster_check_value)
-  }]
+    }
+  ]
+
+  # This will match both datadog and datadog-cluster-agent service account names
+  service_account_name = "datadog*"
+
+  partition  = join("", data.aws_partition.current[*].partition)
+  account_id = join("", data.aws_caller_identity.current[*].account_id)
+
+  iam_role_arn = "arn:${local.partition}:iam::${local.account_id}:role/${module.this.id}-${trimsuffix(local.service_account_name, "*")}@${var.kubernetes_namespace}"
+
+  set_datadog_irsa = var.iam_role_enabled ? [
+    {
+      name  = "clusterAgent.rbac.serviceAccountAnnotations.eks\\.amazonaws\\.com/role-arn"
+      type  = "string"
+      value = local.iam_role_arn
+    },
+    {
+      name  = "agents.rbac.serviceAccountAnnotations.eks\\.amazonaws\\.com/role-arn"
+      type  = "string"
+      value = local.iam_role_arn
+    },
+  ] : []
+}
+
+module "datadog_configuration" {
+  source = "../../datadog-configuration/modules/datadog_keys"
+
+  global_environment_name = null
```

Comment:
Why are we passing null here? seems like leaving it empty and thus the default of `gbl` would be cleaner

Reply:
Since we're getting rid of global, it was deployed to the primary region

Reply:
That's a good point tho. Perhaps i should default the module to omit the environment so it searches in the default environment